x/concentrated-liquidity/position.go

```diff
@@ -94,3 +95,54 @@ func (k Keeper) setPosition(ctx sdk.Context,
 	key := types.KeyPosition(poolId, owner, lowerTick, upperTick)
 	osmoutils.MustSet(store, key, position)
 }
+
+func (k Keeper) setRangePosition(ctx sdk.Context,
+	poolId uint64,
+	owner sdk.AccAddress,
+	lowerTick, upperTick int64,
+	rangePosition *model.RangePosition,
+) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.KeyRangePosition(poolId)
+	osmoutils.MustSet(store, key, rangePosition)
+}
+
+func (k Keeper) RangePositionIterator(ctx sdk.Context,
+	poolId uint64,
+) sdk.Iterator {
+	store := ctx.KVStore(k.storeKey)
+	return sdk.KVStorePrefixIterator(store, types.KeyRangePosition(poolId))
+}
+
+func (k Keeper) finalizeRangeOrderPositions(ctx sdk.Context, poolId uint64) error {
+	pool, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return err
+	}
+
+	currentTick := pool.GetCurrentTick()
+
+	iterator := k.RangePositionIterator(ctx, poolId)
+	defer iterator.Close()
+
+	// iterate over all range positions and withdraw expired positions
+	for ; iterator.Valid(); iterator.Next() {
```

Comment:
These are the position iterations we are trying to avoid.

We seem to have 2 options:
- figure out the accumulator design Dev mentioned
- Implement a Uni-v3 style range order as our v1. This range order does not expire positions, making it only possible to finalize an order with a separate message

Reply:
I am in favor of the latter for MVP, a user must pull out their position on their own instead of being automatically triggered.

For sure interested in working on the accumulator method for a future release, but dont think we should block on it (there are enough unknowns and testing left to keep us busy as it is imo)